auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailFragment.java

```diff
@@ -297,12 +300,59 @@ public void onFailure(@NonNull Exception e) {
                         } else if (e instanceof FirebaseAuthUserCollisionException) {
                             // Collision with existing user email, it should be very hard for
                             // the user to even get to this error due to CheckEmailFragment.
-                            mEmailInput.setError(getString(R.string.error_user_collision));
+
+                            ProviderUtils.fetchTopProvider(mHelper.getFirebaseAuth(), email).addOnSuccessListener(
+                                    getActivity(),
+                                    new OnSuccessListener<String>() {
+                                        @Override
+                                        public void onSuccess(String provider) {
+                                            Toast.makeText(getContext(),
+                                                           R.string.error_user_collision,
+                                                           Toast.LENGTH_LONG)
+                                                    .show();
+
+                                            if (provider == null) {
+                                                throw new IllegalStateException(
+                                                        "User has no providers even though we got a FirebaseAuthUserCollisionException");
+                                            } else if (EmailAuthProvider.PROVIDER_ID.equalsIgnoreCase(
+                                                    provider)) {
+                                                getActivity().startActivityForResult(
+                                                        WelcomeBackPasswordPrompt.createIntent(
```

Comment:
@samtstern One last note before you merge: do we want to pass the user's password to the `WelcomeBackPasswordPrompt`? It could make the sign-in faster and less annoying to the user since I doubt they want to type their password in yet again. 😄